redash/handlers/dashboards.py

```diff
@@ -49,18 +77,45 @@ def get(self, dashboard_slug=None):
             response['public_url'] = url_for('redash.public_dashboard', token=api_key.api_key, org_slug=self.current_org.slug, _external=True)
             response['api_key'] = api_key.api_key
 
+        last_change = models.Change.get_latest(object_id=dashboard.id, object_type=models.Dashboard.__name__)
+        if last_change:
+            response['latest_version'] = last_change.id
+
         return response
 
     @require_permission('edit_dashboard')
     def post(self, dashboard_slug):
         dashboard_properties = request.get_json(force=True)
         # TODO: either convert all requests to use slugs or ids
         dashboard = models.Dashboard.get_by_id_and_org(dashboard_slug, self.current_org)
+
+        # check access permissions
+        if self.current_user.id != dashboard.user.id:
```

Comment:
We should use [`is_admin_or_owner`](https://github.com/getredash/redash/blob/master/redash/permissions.py#L64) here (this is the helper I mentioned in another comment).